TODO: 
---

- [ ] Fix the try_next function. There are no errors but there is no response generated